### ISSUE TITLE
CraftQueue attempting to use potion while having already started a craft

### ISSUE THREAD

**Issue by GingerbreadJ** [Athavar/Athavar.FFXIV.Plugin#20]
Hi - first off, I want to say how much of a godsend this utility is when crafting, especially for collectibles.

One problem I've run into (I don't know if it's new since the patch):
My Cunning Craftsman's Draught ran out after finishing a craft, and the queuer attempted to apply a new potion _after_ it had started the next craft, which both stalled the crafting queue and resulted in that particular craft failing. Would it be possible to check the order of operations?

Also, a request: would it be possible for us to set a threshold at which the queuer will auto-repair gear? I noticed some of my crafts failed and/or weren't HQ even though the sim said they would be, and I suspect it was due to gear breaking in the middle of a craft.

Thank you for your time!

**Comment by Athavar** [comment]
Gear will only break after finishing a craft. Non-HQ can also happen because of a bad timing of excellence/poor-condition.
The simulation is only testing the craft in a linear way and don't contains condition. And skills that have a probability of success have in the simulation a 100% success chance.

For each craft different steps are processed before the crafting begins:
- SwitchToJob
- EnsureRepair
- EnsureMateriaExtracted
- EnsureStats - also applies food and/or medicine if needed
- OpenRecipe
- SelectIngredients
- StartCraft
- WaitSynthesis - is retesting the rotation with current applied buffs. Cancel craft if the simulation fails. (No material is consumed if a craft is canceled before the first skill is executed) 
- DoRotationAction

One thing that is not included in all simulation tests are crafting effecting FC-buffs .

**Comment by GingerbreadJ** [comment]
Thank you for the reply!

Upon checking, what _actually_ seems to be happening (at least with my current round of crafting) is that it is failing to use the potion Cunning Craftsman's Draught HQ. It runs fine if I manually drink it beforehand, but for some reason it gives me the error '[Athavar's Toolbox]Fail to use potion. Is perhaps on CD.' despite both having the potion and it not being on cooldown.

(I still have no idea why it _started_ a craft and _then_ tried to apply the potion that one time, but Code Is Weird and there was probably a once-off bug or lag delay or something.)

The items that turned out NQ rather than HQ were Grade 5 Vitality Alkahests, and I wasn't using any food or potions for those. While I'd not think it unusual if just a few batches turned out NQ, all 60 of the crafts turned out NQ when both the simulator and Teamcraft expected them to turn out HQ with the selected macro, which is just... odd. Could 100% be user error and I selected the wrong macro after looking at  the right one and switching between crafts though.

**Comment by Athavar** [comment]
The error '[Athavar's Toolbox]Fail to use potion. Is perhaps on CD.' only occurs if the plugin fails to use the potion ([source](https://github.com/Athavar/Athavar.FFXIV.Plugin/blob/c65b35e24c08a6a4ee9caf29d7c1e6a1f3eba7d3/src/Athavar.FFXIV.Plugin.CraftQueue/CraftingJob.cs#L390C27-L392), [souce2](https://github.com/Athavar/Athavar.FFXIV.Plugin/blob/c65b35e24c08a6a4ee9caf29d7c1e6a1f3eba7d3/src/Athavar.FFXIV.Plugin.Common/Manager/CommandInterface.Inventory.cs#L36)).
It could be a timing problem that the character was in the animation of exiting the crafting mode while try to use the potion. Or before this food was applied and because of server latency this action was delayed. The only solution I can think of would be to increase the wait duration between these actions in `CraftQueue`. But it's not guaranteed that this is the root cause.

**Comment by Athavar** [comment]
After a few tests and releases, I have removed the check if an item was successfully used. It will now work like a retry and a job should only go to the next step after applying the buffs.

**Comment by GingerbreadJ** [comment]
Thank you so much, it is executing correctly now.

**Comment by colgeee** [comment]
I have the same issue. Initial potion is correctly applied, it would fail to apply potion if it runs out before the craft job has finished.

**Comment by colgeee** [comment]
I think the problem is failing to exit crafting.

**Comment by Athavar** [comment]
Do you mean with "failing to exit crafting" that the craft is not finished?

I could not reproduce the issue. For my test I have used:
- Job: BSM
- To craft:  [Iron Ingot](https://ffxivteamcraft.com/db/en/item/5057)
- Equipment: [Amateur's Cross-pein Hammer](https://ffxivteamcraft.com/db/en/item/2342) (Craftsmanship: 34, Control:19)
- Potion: [Cunning Craftsman's Syrup](https://ffxivteamcraft.com/db/en/item/27959)
- Macro:
```
/ac "Master's Mend" <wait.3>
/ac "Master's Mend" <wait.3>
/ac "Basic Synthesis" <wait.3>
/ac "Basic Synthesis" <wait.3>
/ac "Careful Synthesis" <wait.3>
```
The most annoying thing about this test is the waiting for the potion is run out.

If this happens again, can you please take screenshots from it.

**Comment by colgeee** [comment]
I can get it to fail quite easily. It fails once about every 2nd or 3rd potion. What I've observed is after finishing the current craft, instead of standing up and drinking potion, it immediately goes to craft the next craft. So my assumption was there needs some more delay to make sure standing up is executed.

However, there seem to be some checks in place to not craft without the assigned potion so it stays crafting without using any action. I was able to temporarily mitigate the issue by just quitting the stalled craft (items gets refunded since no action was performed) and it'll drink potion and continue crafting.
